projects/plugins/jetpack/modules/stats/class-jetpack-stats-dashboard-widget.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Adds the Jetpack stats widget to the WordPress dashboard.
+ *
+ * @package jetpack
+ */
+
+use Automattic\Jetpack\Assets\Logo as Jetpack_Logo;
+use Automattic\Jetpack\Status;
+
+/**
+ * Class that adds a adds the Jetpack stats widget to the WordPress dashboard.
+ */
+class Jetpack_Stats_Dashboard_Widget {
+
+	/**
+	 * Indicates whether the class initialized or not.
+	 *
+	 * @var bool
+	 */
+	private static $initialized = false;
+
+	/**
+	 * Initialize the class by calling the setup static function.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		if ( ! self::$initialized ) {
+			self::$initialized = true;
+			self::wp_dashboard_setup();
+		}
+	}
+
+	/**
+	 * Sets up the Jetpack Stats widget in the WordPress dashboard.
+	 */
+	public static function wp_dashboard_setup() {
+		if ( Jetpack::is_connection_ready() ) {
+			add_action( 'jetpack_dashboard_widget', array( __CLASS__, 'dashboard_widget_footer' ), 999 );
+		}
+
+		if ( has_action( 'jetpack_dashboard_widget' ) ) {
+			$jetpack_logo = new Jetpack_Logo();
+			$widget_title = sprintf(
+				// translators: Placeholder is a Jetpack logo.
+				__( 'Stats by %s', 'jetpack' ),
+				$jetpack_logo->get_jp_emblem( true )
+			);
+
+			// Wrap title in span so Logo can be properly styled.
+			$widget_title = sprintf(
+				'<span>%s</span>',
+				$widget_title
+			);
+
+			wp_add_dashboard_widget(
+				'jetpack_summary_widget',
+				$widget_title,
+				array( __CLASS__, 'dashboard_widget' )
+			);
+			wp_enqueue_style( 'jetpack-dashboard-widget', plugins_url( 'css/dashboard-widget.css', JETPACK__PLUGIN_FILE ), array(), JETPACK__VERSION );
+			wp_style_add_data( 'jetpack-dashboard-widget', 'rtl', 'replace' );
+
+			// If we're inactive and not in offline mode, sort our box to the top.
+			if ( ! Jetpack::is_connection_ready() && ! ( new Status() )->is_offline_mode() ) {
+				global $wp_meta_boxes;
+
+				$dashboard = $wp_meta_boxes['dashboard']['normal']['core'];
+				$ours      = array( 'jetpack_summary_widget' => $dashboard['jetpack_summary_widget'] );
+
+				$wp_meta_boxes['dashboard']['normal']['core'] = array_merge( $ours, $dashboard );
+			}
+		}
+	}
+
+	/**
+	 * Fires when the dashboard is loaded.
+	 *
```

Comment:
I wonder if it may make sense to be more verbose here, or in the docblock for the action. Why do we rely on just an action here? What do we expect to be hooked to that action? It may save folks some time in the future, when trying to understand why this is built that way (we don't just add a footer, we also rely on the action to hook the actual graph).

Reply:
Took a stab at this.